### PR TITLE
Article on ackermann function

### LINKED ISSUE #426: Article on Ackermann Function
## 📚 Documentation

Article on Ackermann Function
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes

### PR DESCRIPTION
**Pull request by satcasm**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Article on ackermann function

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Fixes #426

### CODE REVIEW

File: docs/mathematics/ackermann-function.mdx [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+#### Output:
+
+```
+Result: 4
+```
+
+## Complexity Analysis
+
+- Time  complexity - `O(mA(m, n))`
+- Space complexity - `O(m)`
```
Comment (HarshCasper) [inline docs/mathematics/ackermann-function.mdx:133]: What does `m` and `n` means? Add a note, please

File: docs/mathematics/ackermann-function.mdx [hunk 2]
```diff
@@ -0,0 +1,138 @@
+---
+title: Ackermann Function
+slug: /mathematics/Ackermann-Function
+---
+
+## Introduction
+
+Ackermann function is one of the simplest examples of a total computable function,
+that is not primitive recursive.
```
Comment (HarshCasper) [inline docs/mathematics/ackermann-function.mdx:9]: Wrap all of this in a single line. Also add a better introduction

File: docs/mathematics/ackermann-function.mdx [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+
+## Explanation
+
+It is basically a function with two non-negative integers as arguments.
+
+Ackermann function can be declared as:
```
Comment (HarshCasper) [inline docs/mathematics/ackermann-function.mdx:15]: Give a line break

File: docs/mathematics/ackermann-function.mdx [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+```
+A(0, j)=j+1 for j ≥ 0
+A(i, 0)=A(i-1, 1) for i > 0
+A(i, j)=A(i-1, A(i, j-1)) for i, j > 0
+```
+
+Let's have few examples:
```
Comment (HarshCasper) [inline docs/mathematics/ackermann-function.mdx:22]: Give a line break

File: docs/mathematics/ackermann-function.mdx [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+A(i, j)=A(i-1, A(i, j-1)) for i, j > 0
+```
+
+Let's have few examples: 
+```
+For i=1,j=2
+Output=4
+
+For i=3,j=3
+Output=61
```
Comment (HarshCasper) [inline docs/mathematics/ackermann-function.mdx:28]: What does `i` and `j` mean here and above?

File: docs/mathematics/ackermann-function.mdx [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+```
+Ackermann_Function(int i, int j):
+  - Checking whether `i` is `0`,if yes 
+    then applying the associated condition.
+  - Checking whether `i` is greater than `0` and 
+    `j` is `0` , if yes then calling the function
+    with `m-1` and `1`.
+  - Checking whether `i` and `j` are greater than `0`, if yes
+    then applying the associated condition.
+```
```
Comment (HarshCasper) [inline docs/mathematics/ackermann-function.mdx:42]: Don't put pseudo code here. Instead write proper steps describing your code

File: docs/mathematics/ackermann-function.mdx [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+
+## Complexity Analysis
+
+- Time  complexity - `O(mA(m, n))`
+- Space complexity - `O(m)`
+
+## Credits
+
+- [Ritvij Kumar Sharma](https://github.com/ritvij14) contributed the [Dart implementation](https://github.com/TesseractCoding/NeoAlgo/blob/master/Dart/maths/Ackermann_function.dart)
+- [Rajiv Singh](https://github.com/iamrajiv) contributed the [C++ implementation](https://github.com/TesseractCoding/NeoAlgo/blob/master/C-Plus-Plus/math/Ackermann_function.cpp)
```
Comment (HarshCasper) [inline docs/mathematics/ackermann-function.mdx:138]: Do a `git blame` to see who wrote the code. Rajiv has no implemented it.

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: docs/mathematics/ackermann-function.mdx (hunk 7 above)
Comment (satcasm) [reply docs/mathematics/ackermann-function.mdx]: Done.

**Comment by satcasm** [comment]
Accidentally deleted someone's else file. Opening a new PR from another branch.
Sorry for the inconvenience.

**Pull request closed without merging**